src/dirwatch/dirwatch.c

```diff
@@ -165,7 +165,8 @@ static void
 make_clone(xen_interface_t* xen, domid_t* cloneID, uint16_t vlan, char** clone_name)
 {
     char* command = g_malloc0(snprintf(NULL, 0, CLONE_CMD, clone_script, domain_name, vlan, domain_config) + 1);
-    sprintf(command, CLONE_CMD, clone_script, domain_name, vlan, domain_config);
+    const size_t sz = sizeof(command);
```

Comment:
sz == 8 on x64. This is not you expect to see.

Reply:
ya, size_t various with machine. It is 32 bit for 32 bit machine and 64 bit for 64 bit machine.
So i should use g_strdup_printf() instead of sprintf.

Reply:
`g_strdup_printf()` is preferred, yes

Reply:
```
gchar *s;
s = g_strdup_printf(CLONE_CMD, clone_script, domain_name, vlan, domain_config);
printf("** RUNNING COMMAND: %s\n", s);
g_free(s);
```
Is this okay? I am a newbie to glib library.

Reply:
Yes.